src/preamble.js

```diff
@@ -569,16 +569,18 @@ function UTF8ToString(ptr) {
 //   outU8Array: the array to copy to. Each index in this array is assumed to be one 8-byte element.
 //   outIdx: The starting offset in the array to begin the copying.
 //   maxBytesToWrite: The maximum number of bytes this function can write to the array. This count should include the null
-//                    terminator, i.e. if maxBytesToWrite=1, only the null terminator will be written and nothing else.
+//                    terminator (if it's not explicitly disabled), i.e. if maxBytesToWrite=1, and dontWritezero=false,
+//                    then only the null terminator will be written and nothing else.
 //                    maxBytesToWrite=0 does not write any bytes to the output, not even the null terminator.
+//   dontWriteZero: (optional, defaults to false) If true, doesn't write final null terminator
 // Returns the number of bytes written, EXCLUDING the null terminator.
 
-function stringToUTF8Array(str, outU8Array, outIdx, maxBytesToWrite) {
+function stringToUTF8Array(str, outU8Array, outIdx, maxBytesToWrite, dontWriteZero) {
   if (!(maxBytesToWrite > 0)) // Parameter maxBytesToWrite is not optional. Negative values, 0, null, undefined and false each don't write out any bytes.
     return 0;
-
+  dontWriteZero = dontWriteZero === true;
```

Comment:
just did for good measure, but guess you're right... only used for 2 checks in that function, will remove...